### PR TITLE
Don't push if an user account has expired

### PR DESCRIPTION
**Pull request by MatMaul**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog).
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by clokep** [comment]
@MatMaul Looks like this is having some issues with CI. I'm not sure it is from this branch or not, might want to just merge in develop first and see if that fixes it?

File: changelog.d/8353.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Don't push if an user account has expired.
```
Comment (anoadragon453) [inline changelog.d/8353.bugfix:1]:
```suggestion
Don't send push notifications to expired user accounts.
```

File: synapse/storage/databases/main/registration.py [hunk 2]
```diff
@@ -116,6 +116,22 @@ async def get_expiration_ts_for_user(self, user_id: str) -> Optional[int]:
...
+    async def is_account_expired(self, user_id: str, current_ts: int) -> bool:
+        """
+        Returns whether an user account is expired.
+
+        Args:
+            user_id: The user's ID
+            current_ts: The current timestamp
+
+        Returns:
+            Deferred[bool]: whether the user account has expired
```
Comment (anoadragon453) [inline synapse/storage/databases/main/registration.py:128]:
```suggestion
            Whether the user account has expired
```

File: synapse/storage/databases/main/registration.py [hunk 3]
```diff
@@ -116,6 +116,22 @@ async def get_expiration_ts_for_user(self, user_id: str) -> Optional[int]:
...
+        """
+        expiration_ts = await self.get_expiration_ts_for_user(user_id)
+        if expiration_ts is not None and current_ts >= expiration_ts:
+            return True
+        return False
```
Comment (anoadragon453) [inline synapse/storage/databases/main/registration.py:133]:
```suggestion
        return expiration_ts is not None and current_ts >= expiration_ts
```

**Review by anoadragon453 — COMMENTED** [review]
Some tiny fixes, otherwise looks good to me :)

**Review by anoadragon453 — APPROVED** [review]
lgtm!

**Pull request merged**